packages/aws-cdk-lib/aws-cognito/lib/user-pool.ts

```diff
@@ -1400,8 +1431,8 @@ export class UserPool extends UserPoolBase {
       throw new Error('To enable email-based MFA, set `email` property to the Amazon SES email-sending configuration.');
     }
 
-    if (props.advancedSecurityMode === AdvancedSecurityMode.OFF) {
-      throw new Error('To enable email-based MFA, set `advancedSecurityMode` to `AdvancedSecurity.ENFORCED` or `AdvancedSecurity.AUDIT`.');
+    if (props.featurePlan === FeaturePlan.LITE && (!props.advancedSecurityMode || props.advancedSecurityMode === AdvancedSecurityMode.OFF)) {
+      throw new Error('To enable email-based MFA, set `featurePlan` to `FeaturePlan.ESSENTIALS` or `FeaturePlan.PLUS`.');
```

Comment:
Thank you for review.

When `featurePlan` is undefined, CDK cannot determine the actual feature plan. It can be Essentials (newly created user plan) or Lite (created before AWS update). So we should assume  Essentials -- enabling email-based MFA should be allowed.

- When the user pool exists and is to be updated, the actual feature plan is Lite. CloudFormation will fail on deploy. The user should specify `featurePlan` to `ESSENTIALS` (or PLUS) explicitly to change the feature plan.
- When the user pool is to be created, the actual feature plan is Essentials. Deployment will succeed.